runtime/router.go

```diff
@@ -163,7 +164,17 @@ func (endpoint *Endpoint) HandleRequest(
 	req := NewServerHTTPRequest(w, r, params, endpoint)
 
 	fn := endpoint.HandlerFn
-	fn(r.Context(), req, req.res)
+
+	ctx := r.Context()
+	_, ok := ctx.Deadline()
+	if !ok {
+		var cancel context.CancelFunc
+		// TODO: (lu) get timeout from endpoint config
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(100)*time.Millisecond)
```

Comment:
this is not a useful default. This needs to be configurable per endpoint.

Landing this breaks edge-gateway.